src/main/java/org/assertj/core/api/ComparatorFactory.java

```diff
@@ -12,21 +12,80 @@
  */
 package org.assertj.core.api;
 
-import static java.lang.Math.abs;
-
+import java.math.BigDecimal;
 import java.util.Comparator;
+import java.util.Objects;
 
 public class ComparatorFactory {
 
   public static final ComparatorFactory INSTANCE = new ComparatorFactory();
 
+  /**
+   * Judge whether the input number is Nan or Infinity
+   * @param number Object of Float or Double class
+   * @param <T> type of expected and precision, which should be the subclass of java.lang.Number and java.lang.Comparable
+   * @return whether the object is Nan or Infinity
+   * @throws java.lang.NullPointerException if number is null
```

Comment:
that's not what this method do

Reply:
> that's not what this method do

I do not understand. This is the javadoc of `isNanOrInfinity` method. Here are the overview of this method:

```java
/**
   * Judge whether the input number is Nan or Infinity
   * @param number Object of Float or Double class
   * @param <T> type of expected and precision, which should be the subclass of java.lang.Number and java.lang.Comparable
   * @return whether the object is Nan or Infinity
   * @throws java.lang.NullPointerException if number is null
   */
  public static <T extends Number & Comparable<T>> boolean isNanOrInfinity(T number){
    // avoid null argument
    if(number instanceof Float){
      return ((Float) number).isNaN() || ((Float) number).isInfinite();
    }else if(Objects.requireNonNull(number) instanceof Double){
      return ((Double)number).isNaN() || ((Double)number).isInfinite();
    }

    return false;

  }
```
Could you explain why this is not what this method do?

Reply:
you are correct, it does throw a NPE when number is null because of `Objects.requireNonNull(number)` but I asked you not to call `Objects.requireNonNull(number)` as `instanceof` checks for null and returning `false` is correct if number is null